### PR TITLE
meson

### PR DESCRIPTION
**Pull request by konradybcio**
also pkgconfig, some format fixes

### CODE REVIEW

File: include/meson.build [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+install_headers('foolib.h')
```
Comment (superna9999) [inline include/meson.build]: Sees you forgot to add the real headers here

**Review by superna9999 — APPROVED** [review]
Please squash patch 2 into 2, and squash `lib: generate pkgconfig` patch into 1 also, otherwise looks good

File: meson.build [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        compiler_cflags += ['-Wformat-signedness',
+                            '-Wduplicated-cond',
+                            '-Wduplicated-branches',
+                            '-Wvla-larger-than=1',
+                            '-Walloc-zero',
+                            '-Wunsafe-loop-optimizations',
+                            '-Wcast-align',
+                            '-Wlogical-op',
+                            '-Wjump-misses-init']
+endif
```
Comment (lumag) [inline meson.build:20]: I start to hate such lists. What was the reason for selecting -Wfoo but not -Wbar? Can we just set -Wall instead? Or -Wall -Wextra?

Comment (konradybcio) [reply meson.build:20]: tbf, I copypasted it from cdba..

File: meson.build [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        ])
+
+# Set advanced compiler flags
+compiler = meson.get_compiler('c')
+
+prefix = get_option('prefix')
+systemd = dependency('systemd')
+systemd_system_unit_dir = systemd.get_variable(
+        pkgconfig : 'systemdsystemunitdir',
+        pkgconfig_define: ['prefix', prefix])
```
Comment (lumag) [inline meson.build]: On non-usr-merge systems the systemd units usually went to `/lib/systemd`, while the prefix was `/usr`. With Makefiles it was easy to override using the command line. Can we retaing the possibility to override systemd units location please?

File: src/lookup.c [hunk 4]
```diff
@@ -11,7 +11,7 @@
 #include <unistd.h>
 
 #include "libqrtr.h"
-#include "logging.h"
+#include <logging.h>
```
Comment (lumag) [inline src/lookup.c]: So libqrtr.h is local, but logging.h is global? I think this is plain wrong. We are not installing the logging.h header, so it should be local too.

File: meson.build (hunk 2 above)
Comment (konradybcio) [reply meson.build:20]: Should I use `warning_level=everything` from meson instead?

File: src/lookup.c (hunk 4 above)
Comment (konradybcio) [reply src/lookup.c]: Yes, you're right

File: meson.build (hunk 2 above)
Comment (lumag) [reply meson.build:20]: I'd prefer to stick to -Wall. And in CDBA too.

File: meson.build (hunk 3 above)
Comment (konradybcio) [reply meson.build]: so, something like `get_option('systemd-prefix')`?

File: meson.build (hunk 2 above)
Comment (konradybcio) [reply meson.build:20]: fwiw https://mesonbuild.com/Builtin-options.html#details-for-warning_level

**Comment by lumag** [comment]
`systemd-unit-prefix`. And a default if the option is not specified.

File: meson.build [hunk 5]
```diff
@@ -5,7 +5,7 @@ project('qrtr',
         license : [ 'BSD-3-Clause'],
         meson_version : '>= 0.64.0', # for feature user options
         default_options : [
-        #     'warning_level=2',	# sets -Wextra
+        'warning_level=everything',
```
Comment (lumag) [inline meson.build]: =1?

Comment (lumag) [reply meson.build]: Or 2?

**Review by lumag — APPROVED** [review]

**Comment by lumag** [comment]
Nit: it would be nice to make qrtr-ns optional and maybe even skip it by default in the followup.

**Comment by konradybcio** [comment]
> Nit: it would be nice to make qrtr-ns optional and maybe even skip it by default in the followup.

that was my idea for the next pr.. and then a new tagged release right after it - sounds good?

**Comment by lumag** [comment]
SGTM

**Pull request merged**